selfhost/typechecker.jakt

```diff
@@ -1055,6 +1141,7 @@ struct Typechecker {
         OptionalNone(type_id) => type_id
         OptionalSome(type_id) => type_id
         ForcedUnwrap(type_id) => type_id
+        Match(expr, match_cases, span, type_id, all_variants_constant) => type_id
```

Comment:
Hm, interesting I tried that previously but had problems with that where if the signature was not 100% the same (same naming an all params) it could not match.
I'll check, maybe I had a different error and I miss-interpreted it.

Reply:
I just tried it out and you can even put the parameters in a different order and it will assign them properly as long as you use the exact same names. :^)
But yes, it should always be possible to fewer parameters than there are.